pkg/process/config/config.go

```diff
@@ -578,6 +596,18 @@ func getHostname(ddAgentBin string) (string, error) {
 	return hostname, err
 }
 
+func getHostnameFromAgent(ddAgentClient pb.AgentClient) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+	reply, err := ddAgentClient.GetHostname(ctx, &pb.HostnameRequest{})
+
+	if err != nil {
+		return "", err
+	}
+
+	return reply.Hostname, err
```

Comment:
nit: Since you're returning pretty much the same `(string, error)` tuple to the caller, you might be able to just do:
```go
return ddAgentClient.GetHostname(ctx, &pb.HostnameRequest{})
```